src/parsers/utils/fortran.js

```diff
@@ -0,0 +1,127 @@
+import { regex } from "./settings";
+
+const typeParsers = {
+    [Number]: (value) => parseFloat(value),
+    [String]: (value) => value,
+    [Boolean]: (value) => value === "true",
+};
+
+export const FortranParserMixin = (superclass) =>
+    class extends superclass {
+        /**
+         * Parses Fortran namelists and cards data from a string.
+         *
+         * @summary Parses Fortran namelists and cards data from a QE input file string.
+         * @param {String} content - The text to parse.
+         * @throws {Error} If no namelist data is found in `text`.
+         * @throws {Error} If no cards data is found in `text`.
+         * @returns {Object} An object containing the parsed namelist and cards data. The exact structure of this object will depend on the structure of the namelist and cards data in `text`.
+         */
+        parseNamelists(content) {
+            let output = {};
+            try {
+                output = this.extractNamelistData(content);
+            } catch (err) {
+                throw new Error("Incorrect fortran file");
+            }
+
+            const match = regex.fortran.cards.exec(content);
+            // eslint-disable-next-line prefer-destructuring
+            output.cards = match[0];
+            return output;
+        }
+
+        /**
+         * Extracts pairs from a string data using provided regex pattern and type.
+         * If isArray is set to true, treats the value as an array.
+         *
+         * @param {String} data - The string data to extract pairs from.
+         * @param {RegExp} regexPattern - The regex pattern to use for extracting pairs.
+         * @param {Function | NumberConstructor} type - The type of the value.
+         * @param {Boolean} [isArray=false] - Whether to treat the value as an array.
+         *
+         * @returns {Array} The extracted pairs. Each pair is represented as an array,
+         *                  where the first element is the key and the second element is the value.
+         *                  If isArray is true, the value is an array where the first element
+         *                  is the index of the Fortran array element and the second element is the value.
+         * @throws {Error} If an invalid type is provided.
+         */
+        extractPairs(data, regexPattern, type, isArray) {
```

Comment:
The naming is hard to understand and there are no examples in the docstring and/or tests - wth is a "pair"?? Why is there a "pair" and a "key-value pair" that are separate?? Let's add examples

Reply:
I think it was done in order to shorten the function length and isolate repeated code

Reply:
Changed to KeyValuePair(s)